### ISSUE TITLE
java-static-field-call doesn't detect enum fields

### ISSUE THREAD

**Issue by marcobiscaro2112** [clj-kondo/clj-kondo#2395]
**version**
```
$ clj-kondo --version
clj-kondo v2024.08.29
```

**platform**

native version on macOS 14.6.1 installed via homebrew

**problem**

The `:java-static-field-call` linter doesn't detect static field calls when the field is a Java enum value.

**repro**

Only the first (non-enum) field call is detected:

```
$ cat repro.clj
(def a (java.time.temporal.IsoFields/DAY_OF_QUARTER))
(def b (java.time.temporal.ChronoField/DAY_OF_MONTH))
$ clj-kondo --lint repro.clj
repro.clj:1:8: error: Static fields should be referenced without parens unless they are intended as function calls
linting took 9ms, errors: 1, warnings: 0
```

**config**

N/A

**expected behavior**

The field usage in enums (in this case `ChronoField/DAY_OF_MONTH`) should be detected by the java-static-field-call linter as well.

**Comment by borkdude** [comment]
Thanks!

**Comment by borkdude** [comment]
The issue isn't enum fields, but just that there is no class information for this class built into clj-kondo...

```
$ ls -la resources/clj_kondo/impl/cache/built_in/java/java.time.temporal.
java.time.temporal.IsoFields.transit.json                      java.time.temporal.TemporalAdjusters.transit.json              java.time.temporal.WeekFields.ComputedDayOfField.transit.json
java.time.temporal.Temporal.transit.json                       java.time.temporal.TemporalAmount.transit.json                 java.time.temporal.WeekFields.transit.json
java.time.temporal.TemporalAccessor.transit.json               java.time.temporal.TemporalField.transit.json
java.time.temporal.TemporalAdjuster.transit.json               java.time.temporal.TemporalQuery.transit.json
```

I'll see if I can fix that
